Update to Kubernetes v1.16

mikkeloscar opened the pull request:
https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.16.md#v1160

## Changes

* [x] Drop legacy node labels no longer supported by kubelet (e.g. `kubernetes.io/role`)
* [x] Remove `CustomResourceWebhookConversion` feature flag, must be set to `true` in `v1.16` (Introduced in https://github.com/zalando-incubator/kubernetes-on-aws/pull/2172)
* [x] Remove `CustomResourcePublishOpenAPI` feature flag, must be set to `true` in `v1.16` (https://github.com/zalando-incubator/kubernetes-on-aws/pull/2621)
  * [ ] Fix issue which made it break CRDs: https://github.com/zalando-incubator/kubernetes-on-aws/pull/2627
    * [ ] ~Add `status` validation for `PlatformCredentialsSets`~
    * [x] Ensure `status` validation for `AWSIAMRole`
    * [ ] ~Ensure `status` validation for VPA CRDs~
    * [ ] ~Ensure `status` validation for Stackset/Stack CRDs~
* [x] Make it possible to enable `EndpointSlice` (https://github.com/robscott/k8s-website/blob/4370360fc1c46b4e530dec59f5bb45b2c8b24840/content/en/docs/tasks/administer-cluster/enabling-endpoint-slices.md#enabling-endpoint-slices)
* [x] Update flannel/cni to use new config format (conflist) (https://github.com/zalando-incubator/kubernetes-on-aws/pull/2796)
* [x] Update clc to look for `node.kubernetes.io/node-pool` label instead of `kubernetes.io/node-pool` (https://github.com/zalando-incubator/kubernetes-on-aws/pull/2800)
* [x] Fix/workaround e2e test depending on old node label (https://github.com/kubernetes/kubernetes/blob/224be7bdce5a9dd0c2fd0d46b83865648e2fe0ba/test/e2e/framework/util.go#L1976)
  * Fixed by using the hack: `-allowed-not-ready-nodes=1` when running e2e. Can be improved in `v1.17`.
* [x] Support `apiextensions.k8s.io/v1` in admission-controller
* [x] Update CRDs to `apiextensions.k8s.io/v1`
* [x] Update components which use old APIs like daemonset v1beta1
  * [x] Update cluster autoscaler to not use daemonset v1beta1 api.
* [x] Understand/fix broken e2e test: https://github.com/kubernetes/kubernetes/blob/v1.16.4/test/e2e/apimachinery/webhook.go#L207
* [x] Check for changes in metrics
* [ ] Conditionally enable Pod Topology spread: https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/
* [x] Enable `HPAScaleToZero` by default: https://github.com/kubernetes/kubernetes/pull/74526

mikkeloscar committed:
Update to Kubernetes v1.16

Signed-off-by: Mikkel Oscar Lyderik Larsen <mikkel.larsen@zalando.de>

mikkeloscar committed:
Update to Kubernetes v1.16

Signed-off-by: Mikkel Oscar Lyderik Larsen <mikkel.larsen@zalando.de>

szuecs commented:
@mikkeloscar can you please check if you can enable endpointslices, too?
 `--feature-gates=EndpointSlice=true` on api server and `--controllers=endpointslice` on controller manager

Commit:
chore: drop obsolete feature gate for PodMaxPids

Signed-off-by: Martin Linkhorst <martin.linkhorst@zalando.de>

Commit:
chore: drop obsolete feature gate for PodMaxPids

Signed-off-by: Martin Linkhorst <martin.linkhorst@zalando.de>

szuecs committed:
enhance authClient opentracing

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

mikkeloscar committed:
Update Emergency Access Service to show reason after approval

Signed-off-by: Mikkel Oscar Lyderik Larsen <mikkel.larsen@zalando.de>

Commit:
Wait to serve current requests before terminating api-server

To avoid spiking error rates for clients, take api-server out of
the kubernetes.default endpoint and then wait while the current
requests are being served

Signed-off-by: Muaaz Saleem <muhammad.muaaz.saleem@zalando.de>

szuecs committed:
fix kubernetes dataclients refresh

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

zalando-teapot-robot committed:
Merge dev to dev-to-kube-1.16

mikkeloscar committed:
Merge pull request #2987 from zalando-incubator/dev-to-kube-1.16

dev to kube-1.16

mikkeloscar committed:
Merge pull request #2972 from zalando-incubator/master-graceful-shutdown

Shutdown masters more gracefully

mikkeloscar committed:
Enable HPAScaleToZero by default

Signed-off-by: Mikkel Oscar Lyderik Larsen <mikkel.larsen@zalando.de>

zalando-teapot-robot committed:
Merge dev to dev-to-kube-1.16

zalando-teapot-robot committed:
Merge dev to dev-to-kube-1.16

zalando-teapot-robot committed:
Merge dev to dev-to-kube-1.16

zalando-teapot-robot committed:
Merge dev to dev-to-kube-1.16

mikkeloscar committed:
Merge pull request #2998 from zalando-incubator/dev-to-kube-1.16

dev to kube-1.16

zalando-teapot-robot committed:
Merge dev to dev-to-kube-1.16

zalando-teapot-robot committed:
Merge dev to dev-to-kube-1.16

zalando-teapot-robot committed:
Merge dev to dev-to-kube-1.16

arjunrn committed:
Merge pull request #3009 from zalando-incubator/dev-to-kube-1.16

dev to kube-1.16

aermakov-zalando committed:
Update 1.16 AMI

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

mikkeloscar committed:
Merge pull request #3017 from zalando-incubator/update-1-16-ami

Update 1.16 AMI

zalando-teapot-robot committed:
Merge dev to dev-to-kube-1.16

mikkeloscar committed:
Drop allow non-resource URL rule from webhook

Signed-off-by: Mikkel Oscar Lyderik Larsen <mikkel.larsen@zalando.de>

zalando-teapot-robot committed:
Merge dev to dev-to-kube-1.16

mikkeloscar committed:
Merge pull request #3018 from zalando-incubator/dev-to-kube-1.16

dev to kube-1.16

mikkeloscar committed:
Merge pull request #3020 from zalando-incubator/webhook-drop-non-resource-urls

Webhook drop non resource urls

aermakov-zalando committed:
CA: disable --scale-down-delay-after-add

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

aermakov-zalando committed:
Merge pull request #3021 from zalando-incubator/disable-scale-down-delay

Disable scale down delay

aermakov-zalando commented:
👍

mikkeloscar commented:
:+1:
